sound/soc/sof/sof-pci-dev.c

```diff
@@ -35,8 +35,28 @@ static int sof_pci_debug;
 module_param_named(sof_pci_debug, sof_pci_debug, int, 0444);
 MODULE_PARM_DESC(sof_pci_debug, "SOF PCI debug options (0x0 all off)");
 
+const char *sof_override_tplg_name;
+
 #define SOF_PCI_DISABLE_PM_RUNTIME BIT(0)
 
+static int sof_tplg_cb(const struct dmi_system_id *id)
+{
+	sof_override_tplg_name = (const char *)id->driver_data;
+	return 1;
+}
+
+static const struct dmi_system_id sof_tplg_table[] = {
+	{
+		.callback = sof_tplg_cb,
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_FAMILY, "Google_Volteer"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Terrador"),
+		},
+		.driver_data = ("sof-tgl-max98373-rt5682-ssp2.tplg"),
```

Comment:
I'd like the topology name to be less ambiguous, here we don't know if ssp2 applies to the amplifier or the codec.

Also IIRC we started with the headset codec first, then the amplifier, is this too lake to change the topology names for SoundWire TGL?

It'd make more sense to me to have rt5682-sspN-max98373-sspM, that we know exactly what goes where.
It'd be similar to the SoundWire topologies where we list the devices starting with the link0 which happens to be the headset.

Reply:
1. I have updated tplg name appending ssp id's next to amp names.  yes, this way no ambiguity in tplg names. Thank you.

2. on changing to soudnwire TGL topologies:  we only have one SKU as of now.. can we leave it for now and do when necessary arises and along with other changes?